File: SatpJ Back/src/main/java/com/satpj/project/modelo/usuario/Usuario.java

package com.satpj.project.modelo.usuario;

import java.util.List;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Inheritance;
import javax.persistence.InheritanceType;
import javax.persistence.OneToMany;
import javax.persistence.Table;
import javax.validation.constraints.Email;
import javax.validation.constraints.NotNull;


import com.fasterxml.jackson.annotation.JsonIgnore;
import com.satpj.project.modelo.alerta.AlertaUsuario;
import com.satpj.project.modelo.horario.Horario;
import com.satpj.project.modelo.sesion_terapia.SesionUsuario;

/**
 * Entidad usuario Usuario de la plataforma
 */
@Getter
@Setter
@Entity
@Table(name = "usuario")
@Inheritance(strategy = InheritanceType.JOINED)
public class Usuario {
	
	@Id
	@Column(name = "usuario_id")
	private String id;

	/* Son las sesiones de Terapia del Usuario */
	@OneToMany(mappedBy = "usuario")
	@JsonIgnore
	private List<SesionUsuario> sesiones;

	@OneToMany(mappedBy = "usuario")
	@JsonIgnore
	private List<AlertaUsuario> alertasUsuario;

	@OneToMany(mappedBy = "usuario")
	@JsonIgnore
	private List<Horario> horarios;

	@NotNull(message = "El Documento de Identidad es obligatorio")
	@Column(name = "documento", nullable = false, length = (500))
	private String documento;

	@NotNull(message = "El Tipo de Documento de Identidad es obligatorio")
	@Column(name = "tipo_documento", nullable = false)
	private String tipoDocumento;

	@NotNull(message = "El Nombre es obligatorio")
	@Column(name = "nombre", nullable = false)
	private String nombre;

	@NotNull(message = "El Apellido es obligatorio")
	@Column(name = "apellido", nullable = false)
	private String apellido;

	@NotNull(message = "El Email es obligatorio")
	@Email(message = "El Email es invalido")
	@Column(name = "email", nullable = false)
	private String email;

	@NotNull(message = "El Telefono es obligatorio")
	@Column(name = "telefono", nullable = false)
	private String telefono;

	@NotNull(message = "El Tipo de Usuario es obligatorio")
	@Column(name = "tipo_usuario", nullable = false)
	private String tipoUsuario;

	@Column(name = "direccion", nullable = true)
	private String direccion;


	/*
	 * Es un elemento que contiene la clave y la fecha de la sesión actual del
	 * Usuario
	 */
	@Column(name = "info_sesion")
	private String infoSesion;

}

File: satpj_front_end_web/lib/src/model/usuario.dart
import 'dart:convert';

import 'package:satpj_front_end_web/src/model/alerta.dart';
import 'package:satpj_front_end_web/src/model/alerta_usuario.dart';
import 'package:satpj_front_end_web/src/model/horario.dart';
import 'package:satpj_front_end_web/src/model/sesion_terapia.dart';
import 'package:satpj_front_end_web/src/model/sesion_usuario.dart';

List<Usuario> usuarioFromJson(String str) =>
    List<Usuario>.from(json.decode(str).map((x) => Usuario.fromJson(x)));

String usuarioToJson(List<Usuario> data) =>
    json.encode(List<dynamic>.from(data.map((x) => x.toJson())));

class Usuario {
    
    int id;
    List<SesionUsuario> sesiones;
    List<AlertaUsuario> alertasUsuario;
    List<Horario> horarios;
    String documento;
    String tipoDocumento;
    String nombre;
    String apellido;
    String email;
    String telefono;
    String direccion;
    String tipoUsuario;
    String infoSesion;

    Usuario({
      this.id,
      this.sesiones,
      this.alertasUsuario,
      this.horarios,
      this.documento,
      this.tipoDocumento,
      this.nombre,
      this.apellido,
      this.email,
      this.telefono,
      this.tipoUsuario,
      this.infoSesion,
      this.direccion
    });
    
  factory Usuario.fromJson(Map<String, dynamic> json) {
    var list1 = json['sesiones'] as List;
    List<SesionUsuario> lsesiones = list1.map((i) => SesionUsuario.fromJson(i)).toList();
    var list2 = json['alertasUsuario'] as List;
    List<AlertaUsuario> lalertasUsuario = list2.map((i) => AlertaUsuario.fromJson(i)).toList();
    var list3 = json['horarios'] as List;
    List<Horario> lhorarios = list3.map((i) => Horario.fromJson(i)).toList();

    return Usuario(
      id: json["id"],
      sesiones: lsesiones,
      alertasUsuario: lalertasUsuario,
      horarios: lhorarios,
      documento: json["documento"],
      tipoDocumento: json["tipoDocumento"],
      nombre: json["nombre"],
      apellido: json["apellido"],
      email: json["email"],
      telefono: json["telefono"],
      tipoUsuario: json["tipoUsuario"],
      infoSesion: json["infoSesion"],
    );
  }

  Map<String, dynamic> toJson() => {
        "id": id,
        "sesiones": jsonEncode(sesiones),
        "alertasUsuario": jsonEncode(alertasUsuario),
        "horarios": jsonEncode(horarios),
        "documento": documento,
        "tipoDocumento": tipoDocumento,
        "nombre": nombre,
        "apellido": apellido,
        "email": email,
        "telefono": telefono,
        "tipoUsuario": tipoUsuario,
        "infoSesion": infoSesion,
      };
}